src/pineko/evolve.py

```diff
@@ -371,28 +357,34 @@ def xgrid_reshape(full_operator):
     # To compute the alphas values we are first reverting the factorization scale shift
     # and then obtaining the renormalization scale using xir.
     ren_grid2 = xir * xir * mur2_grid
+    # TODO: double-check that getting `nf` from the first Operator is always correct
```

Comment:
I think "by chance" this is correct, because we have only two kind of datasets in game: LHC data, which is always in nf=5 (and so is the first EKO), OR nFONLL grids, which are at at a fixed nf by construction (and so is the first EKO). This would not work for aFONLL since the nf in the observable and alpha_s would need to be in sync (alphas in different flavors is an explicit player in FONLL). The true story is as usual more complicated: actually the grid = PineAPPL should support evolution points, since it is the observable which dictates the evolution point (both for PDF=eko and alpha_s) - we just implicitly assume the above categorization.